[DDP][BE][Docs] Clarify checkpoint support

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68827
* #68792

Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

rohan-varma committed:
[DDP][BE][Docs] Clarify checkpoint support

Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

[ghstack-poisoned]

pritamdamania87 commented on torch/nn/parallel/distributed.py:
```diff
@@ -319,6 +319,15 @@ class DistributedDataParallel(Module, Joinable):
+        checkpointing with :meth:`torch.utils.checkpoint`. DDP will work as
+        expected when there are no unused parameters in the model and each layer
+        is checkpointed at most once (make sure you are not passing
+        `find_unused_parameters=True` to DDP). We currently do not support the
+        case where a layer is checkpointed multiple times, or when there unused
+        parameters in the checkpointed model.
```
This is supported with `static_graph` though right? Maybe we should have an additional comment that if your graph is static this can still be supported?

rohan-varma replied:
It is supported with static graph but it is currently a private API which we haven't documented. My thinking is that in general we shouldn't recommend the use of private APIs in docs and shouldn't say the feature is supported unless we have a public OSS API for it that is in beta state, what do you think?

pritamdamania87 replied:
Ah I see, I didn't realize static graph is still prototype.

rohan-varma committed:
Update on "[DDP][BE][Docs] Clarify checkpoint support"


Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][BE][Docs] Clarify checkpoint support"


Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][BE][Docs] Clarify checkpoint support"


Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][BE][Docs] Clarify checkpoint support"


Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][BE][Docs] Clarify checkpoint support"


Add a note about current checkpoint support with DDP. Note that this
does not include the features enabled with _set_static_graph yet, as it is an
undocumented private API. Once we support static graph as beta feature in OSS
we can add to the note here.

Differential Revision: [D32624957](https://our.internmc.facebook.com/intern/diff/D32624957/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
